packages/manager/apps/pci-ai-endpoints/src/hooks/token/useDatagridColumns.tsx

```diff
@@ -0,0 +1,130 @@
+import { useTranslation } from 'react-i18next';
+import { OsdsChip } from '@ovhcloud/ods-components/react';
+import {
+  DatagridColumn,
+  DataGridTextCell,
+  ActionMenu,
+} from '@ovh-ux/manager-react-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { ODS_CHIP_SIZE } from '@ovhcloud/ods-components';
+import { useQueries } from '@tanstack/react-query';
+import { useGetTokens } from '@/hooks/api/database/token/useToken.hook';
+import { getToken } from '@/data/api/database/token.api';
+import { TokenData } from '@/types/cloud/project/database/token';
+
+export type UseDatagridColumnsProps = {
+  projectId: string;
+  noLimitation: Date;
+  onUpdate: (token: TokenData) => void;
+  onDelete: (token: TokenData) => void;
+};
+
+export const useDatagridColumns = ({
+  projectId,
+  noLimitation,
+  onUpdate,
+  onDelete,
+}: UseDatagridColumnsProps) => {
+  const { t } = useTranslation('token');
+
+  const { data: tokenNames, isLoading: isNamesLoading } = useGetTokens({
+    projectId,
+  });
+
+  const tokenNamesArray: string[] = tokenNames || [];
+
+  const tokenDetailsQueries = useQueries({
+    queries: tokenNamesArray.map((name) => ({
+      queryKey: ['tokens', projectId, name],
+      queryFn: () => getToken({ projectId, name }),
+      enabled: !!projectId,
+    })),
+  });
+
+  const tokenItems: TokenData[] = tokenDetailsQueries
+    .map((query) => query.data)
+    .filter((token): token is TokenData => token !== undefined);
+
+  const isLoading =
+    isNamesLoading || tokenDetailsQueries.some((query) => query.isLoading);
+
+  const columns: DatagridColumn<TokenData>[] = [
+    {
+      id: 'name',
+      label: t('ai_endpoints_token_name'),
+      cell: (token: TokenData) => (
+        <DataGridTextCell>{token.name}</DataGridTextCell>
+      ),
+    },
+    {
+      id: 'description',
+      label: t('ai_endpoints_token_description'),
+      cell: (token: TokenData) => (
+        <DataGridTextCell>
+          {token.description === '' ? '' : token.description || 'N/A'}
```

Comment:
```suggestion
          {token.description === '' ? '' : token.description || 'N/A'}
```

What is the use case ? When is undefined ? Null ? Or both ?